### PR TITLE
Fix unreachable clause where both tmp_plugin and tmp_plugin_instance are non-empty.

### PR DESCRIPTION
**Pull request by qingling128**
ChangeLog: Fix unreachable clause where both tmp_plugin and tmp_plugin_instance are non-empty.

### CODE REVIEW

File: src/aggregation.c [hunk 1]
```diff
@@ -194,17 +194,21 @@ static int agg_instance_create_name(agg_instance_t *inst, /* {{{ */
...
                 sizeof(inst->ident.plugin_instance), "%s-%s", tmp_plugin,
                 AGG_FUNC_PLACEHOLDER);
-    else if (strcmp("", tmp_plugin_instance) != 0)
+    // tmp_plugin is empty and tmp_plugin_instance is non-empty.
+    else if (strcmp("", tmp_plugin) == 0)
       ssnprintf(inst->ident.plugin_instance,
                 sizeof(inst->ident.plugin_instance), "%s-%s",
                 tmp_plugin_instance, AGG_FUNC_PLACEHOLDER);
+    // Both tmp_plugin and tmp_plugin_instance are non-empty.
     else
```
Comment (qingling128) [inline src/aggregation.c]: This `else` clause is not reachable before this fix.

**Comment by qingling128** [comment]
@kwiatrox - Hi, any chance you know what the `clang-format — unexpected status: 500 Internal Server Error` failure is about? If it's intermittent, how can I trigger it again?

**Comment by rpv-tomsk** [comment]
I become happy when I see red sign on clang-format CI.

**Comment by rpv-tomsk** [comment]
@qingling128, that is signal for you: "developers are not welcome here".
JFYI: PR are not merged here, just take a short look on PR history.

**Comment by mrunge** [comment]
@qingling128 Thank you for your patience and your patch. Unfortunately, I don't have an (easy) way to retrigger the check. The simplest way would be to push another change to your branch.

**Comment by qingling128** [comment]
Just pushed some cosmetic changes to re-trigger the builds.

**Comment by mrunge** [comment]
@qingling128 thank you!

Debian build fails with

```
usr/local/Cellar/grpc/1.25.0_1/include/grpcpp/impl/codegen/method_handler_impl.h:170:3: note: in instantiation of member function 'grpc_impl::internal::ServerStreamingHandler<collectd::Collectd::Service, collectd::QueryValuesRequest, collectd::QueryValuesResponse>::RunHandler' requested here

  ServerStreamingHandler(

  ^

collectd.grpc.pb.cc:78:11: note: in instantiation of member function 'grpc_impl::internal::ServerStreamingHandler<collectd::Collectd::Service, collectd::QueryValuesRequest, collectd::QueryValuesResponse>::ServerStreamingHandler' requested here

      new ::grpc::internal::ServerStreamingHandler< Collectd::Service, ::collectd::QueryValuesRequest, ::collectd::QueryValuesResponse>(

          ^

/usr/local/Cellar/protobuf/3.11.1/include/google/protobuf/message_lite.h:401:3: note: 'ByteSize' has been explicitly marked deprecated here

  PROTOBUF_DEPRECATED_MSG("Please use ByteSizeLong() instead")

  ^

/usr/local/Cellar/protobuf/3.11.1/include/google/protobuf/port_def.inc:153:53: note: expanded from macro 'PROTOBUF_DEPRECATED_MSG'

#define PROTOBUF_DEPRECATED_MSG(msg) __attribute__((deprecated(msg)))

                                                    ^

5 errors generated.

make[2]: *** [grpc_la-collectd.grpc.pb.lo] Error 1

make[1]: *** [all] Error 2

make: *** [distcheck] Error 1

The command "make distcheck DISTCHECK_CONFIGURE_FLAGS="--disable-dependency-tracking --enable-debug"" exited with 2.
```

This looks totally unrelated to the change :/

To solve that, we'll need some help from someone with permissions to fix CI.

@octo @mfournier could you have a look please, or provide

**Comment by rpv-tomsk** [comment]
>@qingling128, that is signal for you: "developers are not welcome here".
>JFYI: PR are not merged here, just take a short look on PR history.

I think, you got enough confirmation of that fact.

>I become happy when I see red sign on clang-format CI.

yeah, yeah!
Nobody cares, and that makes me happy again and again!

**Comment by qingling128** [comment]
@mrunge - If the test failures are unrelated, can we disregard those failures?

**Comment by mrunge** [comment]
@qingling128 I would, if I could. Unfortunately, CI failures prevent merging.

**Comment by qingling128** [comment]
@mrunge Thanks for the clarification. Anything else I could do to move this forward? @octo  @mfournier Would retrying the build help (assuming it's intermittent)?

**Comment by mrunge** [comment]
The issue is not intermittent, I already kicked CI and it failed again.
Question is, why this patch fails, and others pass.
@octo and @mfournier are the people with powers on CI, I know @faxm0dem kindly offered his help, but has no permissions yet.

**Comment by qingling128** [comment]
The error seems to be:

```
CXX      grpc_la-collectd.grpc.pb.lo
In file included from collectd.grpc.pb.cc:6:
In file included from ./collectd.grpc.pb.h:42:
/usr/local/Cellar/grpc/1.25.0_1/include/grpcpp/impl/codegen/proto_utils.h:52:23: error: 'ByteSize' is deprecated: Please use ByteSizeLong() instead [-Werror,-Wdeprecated-declarations]
  int byte_size = msg.ByteSize();
                      ^
/usr/local/Cellar/protobuf/3.11.1/include/google/protobuf/message_lite.h:401:3: note: 'ByteSize' has been explicitly marked deprecated here
  PROTOBUF_DEPRECATED_MSG("Please use ByteSizeLong() instead")
  ^
/usr/local/Cellar/protobuf/3.11.1/include/google/protobuf/port_def.inc:153:53: note: expanded from macro 'PROTOBUF_DEPRECATED_MSG'
#define PROTOBUF_DEPRECATED_MSG(msg) __attribute__((deprecated(msg)))
                                                    ^
```

And the error is exclusive to the job with `Xcode: xcode11.2`.

I just saw this commit https://github.com/collectd/collectd/pull/3353 that disabled `Xcode` and it's already merged to master. Let me try rebasing.

**Comment by qingling128** [comment]
@mrunge - Looks like rebasing made the checks passed. Would you mind LGTM this one and getting it merged?

**Comment by mrunge** [comment]
Thank you and sorry for having this loong delay.

**Comment by qingling128** [comment]
@mrunge - No worries. Thank you for the help!

**Pull request merged**